WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/viewholders/LineChartViewHolder.kt

```diff
@@ -0,0 +1,301 @@
+            buildDataSet(context, item.selectedType, mappedEntries)
+        } else {
+            buildEmptyDataSet(context, item.selectedType, item.entries.size)
+        }
+        item.onLineChartDrawn?.invoke(thisWeekDataSet.entryCount)
+
+        val dataSets = mutableListOf<ILineDataSet>()
+        dataSets.add(thisWeekDataSet)
+        dataSets.add(prevWeekDataSet)
+        data = LineData(dataSets)
+
+        axisLeft.apply {
+            valueFormatter = LargeValueFormatter()
+            setDrawGridLines(true)
+            setDrawTopYLabelEntry(true)
+            setDrawZeroLine(false)
+            setDrawAxisLine(false)
+            granularity = 1f
+            axisMinimum = 0f
+            axisMaximum = if (maxYValue < MIN_VALUE) {
+                MIN_VALUE
+            } else {
+                roundUp(maxYValue.toFloat())
+            }
+            setLabelCount(5, true)
+            textColor = ContextCompat.getColor(context, R.color.neutral_30)
+            gridColor = ContextCompat.getColor(context, R.color.stats_bar_chart_gridline)
+            textSize = 10f
+            gridLineWidth = 1f
+        }
+        extraLeftOffset = 8f
+        axisRight.apply {
+            setDrawGridLines(false)
+            setDrawZeroLine(false)
+            setDrawLabels(false)
+            setDrawAxisLine(false)
+        }
+        xAxis.apply {
+            granularity = 1f
+            setDrawAxisLine(false)
+            setDrawGridLines(false)
+
+            setDrawLabels(true)
+            setLabelCount(3, true)
+            position = BOTTOM
+            valueFormatter = LineChartLabelFormatter(thisWeekData)
+
+            removeAllLimitLines()
+        }
+        setPinchZoom(false)
+        setScaleEnabled(false)
+        legend.isEnabled = false
+        setDrawBorders(false)
+
+        val isClickable = item.onLineSelected != null
+        if (isClickable) {
+            setOnChartValueSelectedListener(object : OnChartValueSelectedListener {
+                override fun onNothingSelected() {
+                    item.selectedItemPeriod
+                    item.onLineSelected?.invoke(item.selectedItemPeriod)
+                }
+
+                override fun onValueSelected(e: Entry, h: Highlight) {
+                    val value = (e as? BarEntry)?.data as? String
+                    item.onLineSelected?.invoke(value)
+                }
+            })
+        } else {
+            setOnChartValueSelectedListener(null)
+        }
+
+        isHighlightPerDragEnabled = false
+        isHighlightPerTapEnabled = isClickable
+        val description = Description()
+        description.text = ""
+        this.description = description
+
+        animateX(1000, Easing.EaseInSine)
+
+        invalidate()
+        return item.entries.size
+    }
+
+    private fun buildEmptyDataSet(context: Context, selectedType: Int, count: Int): LineDataSet {
```

Comment:
`selectedType` is never used here. Will we use it later?

Reply:
Thought so, remove it for now, will add it when required.